src/command_modules/azure-cli-container/azure/cli/command_modules/container/_validators.py

```diff
@@ -50,6 +50,54 @@ def validate_image(ns):
                        ns.image)
 
 
+def validate_msi(cmd, namespace):
+    MSI_LOCAL_ID = '[system]'
+    if namespace.assign_identity is not None:
+        identities = namespace.assign_identity or []
+        if not namespace.identity_scope and getattr(namespace.identity_role, 'is_default', None) is None:
+            raise CLIError("usage error: '--role {}' is not applicable as the '--scope' is not provided".format(
+                namespace.identity_role))
+
+        if namespace.identity_scope:
+            if identities and MSI_LOCAL_ID not in identities:
+                raise CLIError("usage error: '--scope'/'--role' is only applicable when assign system identity")
+            # keep 'identity_role' for output as logical name is more readable
+            setattr(namespace, 'identity_role_id', _resolve_role_id(cmd.cli_ctx, namespace.identity_role,
+                                                                    namespace.identity_scope))
+    elif namespace.identity_scope or getattr(namespace.identity_role, 'is_default', None) is None:
+        raise CLIError('usage error: --assign-identity [--scope SCOPE] [--role ROLE]')
+
+
+def _resolve_role_id(cli_ctx, role, scope):
+    import re
+    import uuid
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.cli.core.profiles import ResourceType
+
+    client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_AUTHORIZATION).role_definitions
+    role_id = None
+    if re.match(r'/subscriptions/.+/providers/Microsoft.Authorization/roleDefinitions/',
+                role, re.I):
+        role_id = role
+    else:
+        try:
+            uuid.UUID(role)
+            role_id = '/subscriptions/{}/providers/Microsoft.Authorization/roleDefinitions/{}'.format(
+                client.config.subscription_id, role)
+        except ValueError:
+            pass
+        if not role_id:  # retrieve role id
+            role_defs = list(client.list(scope, "roleName eq '{}'".format(role)))
+            if not role_defs:
+                raise CLIError("Role '{}' doesn't exist.".format(role))
+            elif len(role_defs) > 1:
+                ids = [r.id for r in role_defs]
+                err = "More than one role matches the given name '{}'. Please pick an id from '{}'"
+                raise CLIError(err.format(role, ids))
+            role_id = role_defs[0].id
+    return role_id
```

Comment:
Can you use [parse_resource_id](https://github.com/Azure/msrestazure-for-python/blob/master/msrestazure/tools.py#L104), [resource_id](https://github.com/Azure/msrestazure-for-python/blob/master/msrestazure/tools.py#L183), and [is_valid_resource_id](https://github.com/Azure/msrestazure-for-python/blob/master/msrestazure/tools.py#L228) from msrestazure.tools?

It will likely save you a lot of code/possible bugs and other modules are already using that functionality.

Reply:
This is the same pattern for resolving the role id as in the role and vm modules. I think using parse_resource_id and checking each of the individual components would make it less clear what pattern it is checking.

Reply:
Gotcha, I won't block this PR then.
I'll followup later on to convert all module to use the msrestazure tools.

Reply:
Sounds good. Anything like that you see send me a message and i'll check it out.